dotnetv3/AutoScaling/scenarios/AutoScale_Basics/AutoScale_BasicsTests/AutoScaleMethodsTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using AutoScale_Basics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Amazon.AutoScaling;
+using Amazon.AutoScaling.Model;
+
+namespace AutoScale_Basics.Tests
+{
+    [TestClass()]
+    public class AutoScaleMethodsTests
+    {
+        private string _GroupName = "test-group-name";
+        private AmazonAutoScalingClient _Client = new AmazonAutoScalingClient();
+
+        // the Amazon Resource Name (ARN) of the IAM service linked role.
+        private readonly string _ServiceLinkedRoleArn = "<Enter Value>";
+
+        private readonly string _LaunchTemplateName = "AutoScaleLaunchTemplateTest";
+
+        [TestMethod()]
+        public async Task DeleteAutoScalingGroupTest()
+        {
+            var success = await AutoScaleMethods.DeleteAutoScalingGroupAsync(_Client, _GroupName);
+            Assert.IsTrue(success, "Could not delete the list.");
+        }
+
+        [TestMethod()]
+        public async Task DeleteAutoScalingGroupGroupNonexistentTest()
+        {
+            var success = await AutoScaleMethods.DeleteAutoScalingGroupAsync(_Client, "NonExistentSite");
+            Assert.IsFalse(success, "Could not delete the list.");
+        }
+
+        [TestMethod()]
+        public async Task TerminateInstanceInAutoScalingGroupTest()
+        {
+            var instanceId = string.Empty;
+            var success = await AutoScaleMethods.TerminateInstanceInAutoScalingGroupAsync(_Client, instanceId);
+            Assert.IsTrue(success, "Could not terminate the instance.");
+        }
+
+        [TestMethod()]
+        public async Task TerminateInstanceInAutoScalingGroupNonexistentTest()
+        {
+            var instanceId = string.Empty;
+            var success = await AutoScaleMethods.TerminateInstanceInAutoScalingGroupAsync(_Client, instanceId);
+            Assert.IsFalse(success, "Terminated the instance.");
+        }
+
+        [TestMethod()]
+        public async Task DescribeScalingActivitiesTest()
+        {
+            var activities = await AutoScaleMethods.DescribeAuotoScalingActivitiesAsync(_Client, _GroupName);
+            Assert.IsTrue(activities.Count > 0, "Can't find any auto scaling activities for the group.");
+        }
+
+        [TestMethod()]
+        public async Task SetDesiredCapacityTest()
+        {
+            var success = await AutoScaleMethods.SetDesiredCapacityAsync(_Client, _GroupName, 3);
+            Assert.IsTrue(success, "Couln't set the desired capacity.");
```

Comment:
This assert would be a good spot to check if you're setting the capacity to the actual number, not just if it was successful. I'm not sure on how easy it is to access those properties from here, but something like:
`Assert.Equals(AutoScaleMethods.getDesiredCapacity(), 3);`
after your `Assert.IsTrue` call. 

Also, there's a type in your failure message.

Reply:
Added calls to get and then test the DesiredCapacity value. Thanks for pointing that out.